packages/entity-server/src/routes/hierarchy/EntityDescendantsRoute.ts

```diff
@@ -4,29 +4,32 @@
  */
 
 import { Route } from '@tupaia/server-boilerplate';
+import { formatEntitiesForResponse } from './format';
 import {
   HierarchyRequest,
   HierarchyRequestParams,
   HierarchyRequestBody,
   HierarchyRequestQuery,
-  EntityResponseObject,
+  EntityResponse,
 } from './types';
 
 export type DescendantsRequest = HierarchyRequest<
   HierarchyRequestParams,
-  EntityResponseObject[],
+  EntityResponse[],
   HierarchyRequestBody,
   HierarchyRequestQuery & { includeRootEntity?: boolean }
 >;
 export class EntityDescendantsRoute extends Route<DescendantsRequest> {
   async buildResponse() {
+    const { hierarchyId, entity, allowedCountries, fields, flat } = this.req.ctx;
     const { includeRootEntity = false } = this.req.query;
-    const descendants = await this.req.ctx.entity.getDescendants(this.req.ctx.hierarchyId, {
-      country_code: this.req.ctx.allowedCountries,
+    const descendants = await entity.getDescendants(hierarchyId, {
+      country_code: allowedCountries,
     });
-    const responseEntities = includeRootEntity
-      ? [this.req.ctx.entity].concat(descendants)
-      : descendants;
-    return this.res.ctx.formatEntitiesForResponse(responseEntities);
+    const responseEntities = includeRootEntity ? [entity].concat(descendants) : descendants;
+    if (flat) {
+      return formatEntitiesForResponse(this.req.models, this.req.ctx, responseEntities, flat);
+    }
+    return formatEntitiesForResponse(this.req.models, this.req.ctx, responseEntities, fields);
```

Comment:
I really wanna be able to specify `formatEntitiesForResponse(this.req.models, this.req.ctx, responseEntities, flat || fields)` here but typescript really doesn't seem happy with that as it can't safely infer the type of `flat || fields` 😢